eth/client.go

```diff
@@ -418,7 +419,186 @@ func (c *client) Bond(amount *big.Int, to ethcommon.Address) (*types.Transaction
 		}
 	}
 
-	return c.BondingManagerSession.Bond(amount, to)
+	// Get transcoder pool
+	transcoders, err := c.TranscoderPool()
+	if err != nil {
+		return nil, errors.Wrapf(err, "unable to get transcoder pool")
+	}
+
+	// Get max pool size
+	maxSize, err := c.GetTranscoderPoolMaxSize()
+	if err != nil {
+		return nil, errors.Wrapf(err, "unable to get transcoder pool max size")
+	}
+
+	// Get current delegate
+	delegator, err := c.GetDelegator(sender)
+	if err != nil {
+		return nil, errors.Wrapf(err, "unable to get delegator")
+	}
+
+	// Switching delegate's calculate old delegate positions
+	var oldHints lpTypes.TranscoderPoolHints
+	if delegator.DelegateAddress != to {
+		oldHints = findTranscoderHints(delegator.DelegateAddress, transcoders)
+	}
+
+	currentRound, err := c.CurrentRound()
+	if err != nil {
+		return nil, err
+	}
+
+	ep, err := c.GetTranscoderEarningsPoolForRound(to, currentRound)
```

Comment:
When inserting into the on-chain pool, the transcoder's `delegatedAmount` will be considered which can change during the round, but we're using the transcoder's total stake stored in the earnings pool which is locked in at the beginning of a round. We should be able to use the DelegatedAmount field included in the value returned by `GetDelegator()` called for the delegate here instead of calling `GetTranscoderEarningsPoolForRound()`.

Reply:
fixed in 6c90f60